### PR TITLE
provider/aws: Convert AWS Security Groups to aws-sdk-go

### PR DESCRIPTION
**Pull request by catsby**
Converts AWS Security Group resource to use aws-sdk-go

### CODE REVIEW

File: builtin/providers/aws/resource_aws_security_group.go [hunk 1]
```diff
@@ -313,34 +316,49 @@ func resourceAwsSecurityGroupRuleHash(v interface{}) int {
...
+		if perm.FromPort == nil {
+			fromPort = 0
+		} else {
+			fromPort = *perm.FromPort
+		}
+		if perm.ToPort == nil {
+			toPort = 0
+		} else {
+			toPort = *perm.ToPort
+		}
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_security_group.go]: Sad. But understandable.

File: builtin/providers/aws/resource_aws_security_group.go [hunk 2]
```diff
@@ -383,8 +400,9 @@ func resourceAwsSecurityGroupUpdateRules(
 		os := o.(*schema.Set)
 		ns := n.(*schema.Set)
 
-		remove := expandIPPerms(d.Id(), os.Difference(ns).List())
-		add := expandIPPerms(d.Id(), ns.Difference(os).List())
+		// TODO: re-munge this when test is updated
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_security_group.go]: Leftover TODO?

File: builtin/providers/aws/resource_aws_security_group_test.go [hunk 3]
```diff
@@ -44,16 +46,18 @@ func TestAccAWSSecurityGroup_normal(t *testing.T) {
...
 		defer func() {
 			if e := recover(); e != nil {
+				log.Printf("\n\nbad here!!")
 				err = fmt.Errorf("bad: %#v", group)
 			}
 		}()
 
-		if group.IPPerms[0].SourceGroups[0].Id != group.Id {
+		if *group.IPPermissions[0].UserIDGroupPairs[0].GroupID != *group.GroupID {
+			log.Printf("\n\n---- bad here ----\n")
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_security_group_test.go]: Extra debug log here and L54

File: builtin/providers/aws/resource_aws_security_group.go (hunk 1 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_security_group.go]: I in fact cried here :cry:

File: builtin/providers/aws/resource_aws_security_group_test.go (hunk 3 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_security_group_test.go]: derp

File: builtin/providers/aws/resource_aws_security_group.go (hunk 1 above)
Comment (mitchellh) [inline builtin/providers/aws/resource_aws_security_group.go]: Lifechanging edit:

```go
var fromPort, toPort int
if v := perm.FromPort; v != nil {
    fromPort = *v
}
```

(same for `toPort`).

The zero value for ints is zero, so you don't need to set it to zero. This is specified in the Go language specification so it is safe to code this way.

**Comment by mitchellh** [comment]
LGTM

File: builtin/providers/aws/resource_aws_security_group.go (hunk 1 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_security_group.go]: @mitchellh fantastic, thanks. Cleaned up.

**Pull request merged**